### PR TITLE
W22-5pm-2 Personal Schedule Form Component

### PR DESCRIPTION
**Pull request by tiny-babies**
# Overview

Created Form Component, Storybook, and tests for PersonalSchedules.

# Note

At the time of this PR Comment, PersonalSchedules Entities and Repository has not been merged into main.  Some of the variables in this PR are dependent on how they are defined in Personal Schedules Entities, so this may need to be addressed later if issues arise after PersonalSchedules Entities is merged into main.

# Details

image

### CODE REVIEW

**Review by AlexLaiUCSB — APPROVED** [review]
LGTM

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js
Comment (pconrad) [inline frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js:62]: I wonder whether it would be helpful to use the QuarterSelectorDropdown here.   I think that would result in fewer errors in terms of data entry.

**Review by pconrad — DISMISSED** [review]
I think it might be better to use the SingleQuarterDropdown component instead of entering the quarter in free form text.

In order for personal schedules to work with the rest of the app, the quarter needs to be stored internally in the format YYYYQ (e.g. 20201 for Winter 2020, 20224 for Fall 2022, etc.).  But that's an inconvenient format for humans to enter.  The SingleQuarterDropdown takes care of that.

https://github.com/ucsb-cs156-w22/team04-w22-5pm-courses/blob/main/frontend/src/main/components/Quarters/SingleQuarterDropdown.js

**Comment by pconrad** [comment]
10 points if/when this is merged.

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js
Comment (tiny-babies) [reply frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js:68]: ### New Commit:
Finished - Replaced type text form control with the SingleQuarterDropdown component. Ranges from Winter 2008 to Fall 2024 right now. 
### Future Considerations:
 Future fixes may be to edit the range if necessary, and also the SingleQuarterDropdown component starts from the earliest quarter and ends at the most recent, which is annoying and could also be a future issue to resolve. 
image

**Comment by tiny-babies** [comment]
### New Commit:
Finished - Replaced type text form control with the SingleQuarterDropdown component. Ranges from Winter 2008 to Fall 2024 right now. 
### Future Considerations:
 Future fixes may be to edit the range if necessary, and also the SingleQuarterDropdown component starts from the earliest quarter and ends at the most recent, which is annoying and could also be a future issue to resolve. 
image

**Comment by pconrad** [comment]
Good for 10 points.. just waiting on CI/CD after merging in latest changes from main.

**Pull request merged**
